views/FeedPost.js

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import { StyleSheet, Text, View, Image, TouchableOpacity } from "react-native";
+
+function buildPost(item) {
+  let displayText;
+  switch (item.type) {
+    case "WatchEvent":
+      displayText = `${item.actor.login} ${item.payload.action} watching ${
+        item.repo.name
+      } at ${item.created_at}`;
+      break;
+    case "CreateEvent":
+      displayText = `${item.actor.login} created ${item.payload.ref_type} ${
+        item.payload.ref
+      } in ${item.payload.master_branch} inside the repo ${
+        item.repo.name
+      } at ${item.created_at}`;
+      break;
+
+    case "ForkEvent":
+      displayText = `${item.actor.login} forked ${item.repo.name} of ${
+        item.org.login
+      } at ${item.created_at}`;
+      break;
+    case "PushEvent":
+      displayText = `${item.actor.login} commited ${item.payload.before} to ${
+        item.payload.ref
+      } in ${item.payload.master_branch} inside the repo ${
+        item.repo.name
+      } at ${item.created_at}`;
+      break;
+    case "PullRequestEvent":
+      displayText = `${item.actor.login} created ${item.payload.ref_type} ${
+        item.payload.ref
+      } in ${item.payload.master_branch} inside the repo ${
+        item.repo.name
+      } at ${item.created_at}`;
+      break;
+
+    default:
+      displayText = `${item.actor.login} did something else: ${item.type}`;
+      break;
+  }
+  return displayText;
+}
+
+export default class FeedScreen extends React.Component {
+  
+  /**
+   * 
+   * @param {*} item 
+   * @param {*} userAction user clicks on post or on comment button. Possible values: details, comment
+   */
+  handleListTap(item, userAction='details') {
+    this.props.navigation.navigate("Post", {
+      displayText: buildPost(item),
+      item: item,
+      userAction: userAction,
+    });
+  }
+
+  render() {
+    const displayText = buildPost(this.props.item);
```

Comment:
- [ ]  change item name to 'event'

Reply:
changed to feedEvent, event is too generic imho